libvast/vast/system/explorer.hpp

```diff
@@ -28,13 +28,24 @@
 namespace vast::system {
 
 struct explorer_state {
+  struct event_limits {
+    int64_t total;
+    int64_t per_result;
+  };
+
   static inline constexpr const char* name = "explorer";
 
   explorer_state(caf::event_based_actor* self);
 
   /// Send the results to the sink, after removing duplicates.
   void forward_results(vast::table_slice_ptr slice);
 
+  /// Maximum number of events to output.
+  event_limits limits;
+
+  /// Number of events sent to the source.
+  size_t sent;
```

Comment:
```suggestion
  size_t num_sent;
```

We usually prefix counter variables with `num_`.